sdk/python/kfp/components/component_factory.py

```diff
@@ -493,9 +496,7 @@ def make_input_for_parameterized_container_component_function(
     elif type_annotations.is_output_artifact(annotation):
 
         if type_annotations.is_list_of_artifacts(annotation.__origin__):
-            raise ValueError(
-                'Outputting a list of artifacts from a Custom Container Component is not currently supported.'
-            )
+            return placeholders.OutputListOfArtifactsPlaceholder(name)
```

Comment:
Support output lists of artifacts for primitive components here, even though it is blocked downstream in `base_component.py` logic. All exceptions related to output lists of artifacts originate from `base_component.py`.